lib/lf/base/static_vars.h

```diff
@@ -77,6 +77,12 @@ using StaticVar = Track<unsigned int>;
  * @name Macros for handling diagnostics control variables
  */
 /**@{*/
+
+#define ADDOPTION(uintvar, name, comment)                             \
```

Comment:
Please put extensive doxygen comments here that explain how `ADDOPTION`, `CONTROLDECLARE` and `CONTROLDECLARECOMMENT` work. I would also suggest to create doxygen snippets and include them here, see e.g. `lib/lf/io/gmsh_reader.h` for how snippets are used. (They are defined under `doc/doxygen/snippets`)

Reply:
done. :+1: